examples/experimental/dagster-airlift/dagster_airlift/within_airflow.py

```diff
@@ -59,3 +62,208 @@ def mark_as_dagster_migrating(
             )
             globals_to_update[var] = new_dag
     global_vars.update(globals_to_update)
+
+
+ASSET_NODES_QUERY = """
+query AssetNodeQuery {
+    assetNodes {
+        id
+        assetKey {
+            path
+        }
+        opName
+        jobs {
+            id
+            name
+            repository {
+                id
+                name
+                location {
+                    id
+                    name
+                }
+            }
+        }
+    }
+}
+"""
+
+TRIGGER_ASSETS_MUTATION = """
+mutation LaunchAssetsExecution($executionParams: ExecutionParams!) {
+  launchPipelineExecution(executionParams: $executionParams) {
+    ... on LaunchRunSuccess {
+      run {
+        id
+        pipelineName
+        __typename
+      }
+      __typename
+    }
+    ... on PipelineNotFoundError {
+      message
+      __typename
+    }
+    ... on InvalidSubsetError {
+      message
+      __typename
+    }
+    ... on RunConfigValidationInvalid {
+      errors {
+        message
+        __typename
+      }
+      __typename
+    }
+    ...PythonErrorFragment
+    __typename
+  }
+}
+
+fragment PythonErrorFragment on PythonError {
+  message
+  stack
+  errorChain {
+    ...PythonErrorChain
+    __typename
+  }
+  __typename
+}
+
+fragment PythonErrorChain on ErrorChainLink {
+  isExplicitLink
+  error {
+    message
+    stack
+    __typename
+  }
+  __typename
+}
+"""
+
+# request format
+# {
+#   "executionParams": {
+#     "mode": "default",
+#     "executionMetadata": {
+#       "tags": []
+#     },
+#     "runConfigData": "{}",
+#     "selector": {
+#       "repositoryLocationName": "toys",
+#       "repositoryName": "__repository__",
+#       "pipelineName": "__ASSET_JOB_0",
+#       "assetSelection": [
+#         {
+#           "path": [
+#             "bigquery",
+#             "raw_customers"
+#           ]
+#         }
+#       ],
+#       "assetCheckSelection": []
+#     }
+#   }
+# }
+
+RUNS_QUERY = """
+query RunQuery($runId: ID!) {
+	runOrError(runId: $runId) {
+		__typename
+		...PythonErrorFragment
+		...NotFoundFragment
+		... on Run {
+			id
+			status
+			__typename
+		}
+	}
+}
+fragment NotFoundFragment on RunNotFoundError {
+	__typename
+	message
+}
+fragment PythonErrorFragment on PythonError {
+	__typename
+	message
+	stack
+	causes {
+		message
+		stack
+		__typename
+	}
+}
+"""
+
+
+def compute_fn() -> None:
+    # https://github.com/apache/airflow/discussions/24463
+    os.environ["NO_PROXY"] = "*"
+    dag_id = os.environ["AIRFLOW_CTX_DAG_ID"]
+    task_id = os.environ["AIRFLOW_CTX_TASK_ID"]
```

Comment:
Just for documentation purposes I think it would be worth having a dumb value object with all the airflow context env vars that we depend on. We can also use this for testing.